### PR TITLE
Add binding for LIS2MDL magnetometer

### PR DESCRIPTION
**Pull request by josesimoes**
## Description
- Add library, sample and nuspec for publish and distribuition.

## How Has This Been Tested?
- Testde with MXChip AZ3166 device.

## Types of changes

- [ ] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [x] My code follows the code style of this project (only if there are changes in source code).
- [x] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [x] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [x] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/main/CONTRIBUTING.md) document.
- [x] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: devices/Lis2Mdl/Lis2Mdl.cs [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        /// <summary>
+        /// This holds the magnetic field reading from the sensor.
+        /// Elements index is 0 for X, 1 for Y and 2 for Z.
+        /// </summary>
+        private float[] _magneticFieldReadings = new float[3];
+
+        /// <summary>
+        /// Device I2C Address.
+        /// </summary>
+        public const byte I2cAddress = 0x1E;
```
Comment (Ellerbach) [inline devices/Lis2Mdl/Lis2Mdl.cs:44]: We use `DefaultI2cAddress` as a convention

File: devices/Lis2Mdl/Lis2Mdl.cs [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        public MagneticField MagneticFieldZ
+        {
+            get
+            {
+                GetMagneticField();
+
+                return UnitsNet.MagneticField.FromMilligausses(_magneticFieldReadings[2]);
+            }
+        }
+
```
Comment (Ellerbach) [inline devices/Lis2Mdl/Lis2Mdl.cs:113]: For convenience, shall we add a Vector3 with the raw data in Milligauss in there?

File: devices/Lis2Mdl/Lis2Mdl.cs [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+            _i2c.WriteByte((byte)register);
+            return _i2c.ReadByte();
+        }
+
+        private float GetTemperature()
+        {
+            // read the 2 raw data registers into data array      
+            var tempRaw = ReadInt16(Register.TemperatureLow);
+
+            // TwosComplement(tempRaw);
```
Comment (Ellerbach) [inline devices/Lis2Mdl/Lis2Mdl.cs:167]: why is this commented?

**Review by Ellerbach — COMMENTED** [review]
Nice! Very few comments

File: devices/Lis2Mdl/Lis2Mdl.cs (hunk 2 above)
Comment (josesimoes) [reply devices/Lis2Mdl/Lis2Mdl.cs:113]: and "drop" the unit value? 🤔

**Review by Ellerbach — APPROVED** [review]
Seems perfect! Thanks!

**Pull request merged**